hs_core/receivers.py

```diff
@@ -202,3 +205,20 @@ def hs_update_web_services(sender, **kwargs):
                 settings.HSWS_PUBLISH_URLS,
                 rid
             ), countdown=1)
+
+
+@receiver(pre_delete, sender=User)
+def pre_delete_user_handler(sender, instance, **kwargs):
+    # before delete the user, update the quota holder for all of the user's resources
+    user = instance
+    for res in BaseResource.objects.filter(quota_holder=user):
+        other_owners = None
+        if hasattr(res, 'raccess') and res.raccess is not None:
```

Comment:
Since we are not deleting user, I am not considering making suggestions to optimizing queries.

Reply:
I guess maybe that check isn't necessary. I'm not remembering now why I added that...

Reply:
Added it back in. See test failures here:
http://ci.hydroshare.org:8080/job/hydroshare-pull-requests/7611/

Reply:
It is fine to have that check. However, the reason I think the tests failed when you removed that check is because of the delete order here - the `ResourceAccess` records are deleted before `User `records:
https://github.com/hydroshare/hydroshare/blob/develop/hs_access_control/tests/utilities.py#L35
https://github.com/hydroshare/hydroshare/blob/develop/hs_access_control/tests/utilities.py#L36

If you swap the delete order the tests are likely to pass. Again, I am not suggesting you need to make any changes for this PR.

Reply:
right you are! Thanks Pabitra!